Source/Android/app/src/main/AndroidManifest.xml

```diff
@@ -28,6 +28,7 @@
     <uses-permission
         android:name="android.permission.VIBRATE"
         android:required="false"/>
+    <uses-permission android:name="android.permission.RECORD_AUDIO"/>
```

Comment:
```suggestion
    <uses-permission
        android:name="android.permission.RECORD_AUDIO"
        android:required="false"/>
```

Otherwise Google Play can prevent Dolphin from being installed on devices with no microphone.

Reply:
AFAICT, cubeb doesn't prompt anything and just [fails to initialise](https://github.com/mozilla/cubeb/blob/19fcbefe1a9c5e22f8111af251df27b41658bc77/src/cubeb_aaudio.cpp#L987-L994).

Should we check this permission and ask the user the appropriate permission? If so, should it be done in the C++ code before the initialisation or in the Android code when using the settings toggle?

Reply:
Sorry, I missed your comments until you pushed a new commit now.

Yes, I think we should. It's probably best to do it both when turning on the setting and when the C++ code tries to use it. The former because that makes the purpose of the permission the most clear to the user, and the latter in case the permission has been revoked by the user (or automatically by the OS because the app hasn't been used for a while).